src/components/MediaPlayer.vue

```diff
@@ -0,0 +1,283 @@
+<template>
+  <div class="media">
+    <div class="media__cover">
+      <g-image :src="coverArtSrc" />
+    </div>
+    <div class="media__controls">
+      <div class="media__player">
+        <button
+          class="media__pause-play-button"
+          @click="togglePlay">
+          <span :class="isPlaying ? 'pause-icon' : 'play-icon'" />
+        </button>
+        <div class="media__time-line">
+          <div 
+            :style="{transform: `translateX(-${100 - percentComplete}%)`}"
+            class="media__progress-line"
+          />
+        </div>
+      </div>
+      <div class="media__bottom-area">
+        <span class="media__current-time">{{ formattedCurrentTime }}</span>
+        <div class="media__links">
+          <a 
+            v-if="sharingLink"
+            :href="sharingLink"
+          >Share</a>
+          <a 
+            v-if="downloadLink"
+            :href="downloadLink" 
+          >Download</a>
+          <a 
+            v-if="rssLink"
+            :href="rssLink" 
+            target="_blank"
+          >Subscribe</a>
+        </div>
+      </div>
+    </div>
+    
+    <!-- The actual audio player that is hidden from view -->
+    <audio 
+      controls="controls"
+      ref="player"
+      style="display: none;"
+    >
+      <source :src="audioLink" />
+    </audio>
+  </div>
+</template>
+<script>
+export default {
+  name: 'MediaPlayer',
+  props: {
+    coverArtSrc: {
+      type: String,
+      required: false,
+      default: '',
+    },
+    rssLink: {
+      type: String,
+      required: false,
+      default: '',
+    },
+    downloadLink: {
+      type: String,
+      required: false,
+      default: '',
+    },
+    sharingLink: {
+      type: String,
+      required: false,
+      default: '',
+    },
+    audioLink: {
+      type: String,
+      required: false,
+      default: '',
+    },
+  },
+  data() {
+    return {
+      player: null,
+      isPlaying: false,
+      length: 0,
+      currentTime: 0,
+    }
+  },
+  computed: {
+    percentComplete() {
+			return parseInt(this.currentTime / this.length * 100);
```

Comment:
i dont know why this weird formatting is occurring 😩

Reply:
For some reason it's using tabs instead of spaces. Let me see if I can fix it

Reply:
Thank you!